asapdiscovery-modeling/asapdiscovery/modeling/tests/test_protein_prep.py

```diff
@@ -0,0 +1,284 @@
+
+import pytest
+import yaml
+from asapdiscovery.data.openeye import (
+    load_openeye_cif1,
+    load_openeye_pdb,
+    oechem,
+    save_openeye_pdb,
+)
+from asapdiscovery.data.schema import CrystalCompoundData, CrystalCompoundDataset
+from asapdiscovery.data.testing.test_resources import fetch_test_file
+from asapdiscovery.data.utils import seqres_to_res_list
+from asapdiscovery.modeling.modeling import (
+    add_seqres_to_openeye_protein,
+    align_receptor,
+    mutate_residues,
+    remove_extra_ligands,
+    spruce_protein,
+)
+
+
+@pytest.fixture
+def sars():
+    return fetch_test_file("Mpro-P2660_0A_bound.pdb")
+
+
+@pytest.fixture
+def mers():
+    return fetch_test_file("rcsb_8czv-assembly1.cif")
+
+
+@pytest.fixture
+def ref():
+    return fetch_test_file("reference.pdb")
+
+
+@pytest.fixture
+def loop_db():
+    return fetch_test_file("fragalysis-mpro_spruce.loop_db")
+
+
+# This needs to have a scope of session so that a new tmp file is not created for each test
+@pytest.fixture(scope="session")
+def prepped_files(tmp_path_factory, local_path):
+    if not type(local_path) == str:
+        return tmp_path_factory.mktemp("test_prep")
+    else:
+        local_path = Path(local_path)
+        local_path.mkdir(exist_ok=True)
+        assert local_path.exists()
+        return local_path
+
+
+@pytest.fixture
+def sars_xtal(sars):
+    return CrystalCompoundData(
+        str_fn=str(sars),
+        active_site_chain="A",
+        output_name=Path(sars).stem,
+        lig_chain="A",
+    )
+
+
+@pytest.fixture
+def mers_xtal(mers):
+    return CrystalCompoundData(
+        str_fn=str(mers),
+        active_site_chain="A",
+        output_name=Path(mers).stem,
+        active_site="HIS:41: :A:0: ",
+        lig_chain="B",
+    )
+
+
+@pytest.fixture
+def xtal_dataset(sars_xtal, mers_xtal):
+    xtal_dataset = CrystalCompoundDataset(structures=[sars_xtal, mers_xtal])
+    return xtal_dataset
+
+
+class TestCrystalCompoundDataset:
+    def test_dataset_creation(self, xtal_dataset, sars_xtal, mers_xtal):
+        assert len(xtal_dataset.structures) == 2
+        assert xtal_dataset.structures[0].str_fn == str(sars_xtal.str_fn)
+        assert xtal_dataset.structures[1].str_fn == str(mers_xtal.str_fn)
+
+    def test_dataset_saving(self, xtal_dataset, prepped_files, csv_name="to_prep.csv"):
+        to_prep_csv = prepped_files / csv_name
+        xtal_dataset.to_csv(to_prep_csv)
+        assert to_prep_csv.exists()
+        assert to_prep_csv.is_file()
+
+    def test_dataset_loading(self, xtal_dataset, prepped_files, csv_name="to_prep.csv"):
+        dataset = CrystalCompoundDataset()
+        dataset.from_csv(prepped_files / csv_name)
+        assert dataset == xtal_dataset
+
+        dataset.structures[0].active_site_chain = "B"
+        assert dataset != xtal_dataset
+
+
+class TestProteinPrep:
+    def test_sars_protein_prep(
+        self, sars_xtal, ref, prepped_files, loop_db, ref_chain="A"
+    ):
+        xtal = sars_xtal
```

Comment:
Similar idea here about easy decoupling of test and workflow, we want to test a function or script, If we have a currently optimal prep **script** you can use something like:

```python
@pytest.fixture()
def make_output_dir_and_cleanup():
    # create output dir
    os.makedirs("./outputs", exist_ok=True)
    yield
    # clean up
    shutil.rmtree("./outputs")


@pytest.mark.timeout(400)
@pytest.mark.script_launch_mode("subprocess")
def test_single_target_docking(
    script_runner,     make_output_dir_and_cleanup,
):
    args = [
        "prep-proteins",
        "-o",
        "output"
        arg2,
        arg3,
    ]
    ret = script_runner.run(*args)
    assert ret.success

```

Otherwise you can make this logic a function, test the function and use the function inside of a script if that makes sense?

So like 
```python 
# in module
def  protein_prep():

# in script 
...
protein_prep(...)
...

# in tests 
def  test_protein_prep():

```

Reply:
we discussed making this a `prep_sars` function.